saleor/graphql/order/mutations/order_confirm.py

```diff
@@ -60,6 +62,34 @@ def get_instance(cls, info: ResolveInfo, **data):
             )
         return instance
 
+    @staticmethod
+    def charge_transaction_items(user, app, order, manager):
+        for transaction_item in order.payment_transactions.filter(
+            available_actions__contains=[TransactionAction.CHARGE],
+            authorized_value__gt=0,
+        ):
+            charge_value = transaction_item.authorized_value
```

Comment:
you could just make `min(charge_value, order.total.gross.amount)` instead of `if`

Reply:
Also, you could have multiple transaction_items. So you need to reduce the value that you need to charge, based on the sum of the previously charged value.